plugins/commands/port/command.rb

```diff
@@ -0,0 +1,71 @@
+require "optparse"
+
+module VagrantPlugins
+  module CommandPort
+    class Command < Vagrant.plugin("2", :command)
+      def self.synopsis
+        "displays information about guest port mappings"
+      end
+
+      # @todo support multiple strategies if requested by the community
+      def execute
+        opts = OptionParser.new do |o|
+          o.banner = "Usage: vagrant port [options] [name]"
+          o.separator ""
+          o.separator "Options:"
+          o.separator ""
+
+          o.on("--machine-readable", "Display machine-readable output")
+        end
+
+        # Parse the options
+        argv = parse_options(opts)
+        return if !argv
+
+        @logger.debug("Port command: #{argv.inspect}")
+        with_target_vms(argv, single_target: true) do |vm|
+          vm.action_raw(:config_validate,
+            Vagrant::Action::Builtin::ConfigValidate)
+
+          if vm.state.id != :running
```

Comment:
There is a middleware action for CheckRunning (commented out below), but it's only defined in the virtualbox stack. Should that be moved into the main middleware @mitchellh? If so, we could just use that instead of checking the state id directly.